### DISCUSSION TITLE
LCD Overall Dimensions/Resolution

### DISCUSSION THREAD

**Discussion by davewatson91** [elgatosf/elgato#85] [Help] [answered]
I am trying to create a screensaver for a Streamdeck +, but struggling with the correct resolutions for this (to get it spot on).

The resolutions I received directly from Elgato Support (who then sent me in the direction of GitHub for developer support) were: 

> Thank you for contacting Elgato Technical Support.
>  
> I understand that you're experiencing an issue with finding the correct dimensions for a screensaver on your Elgato Stream Deck +. I'd be happy to assist you.
>  
> While we would not have any templates per say, I was able to find the dimensions needed for creating wallpapers for the Elgato Stream Deck + which should also be the same for screensavers. Hopefully this is some information that helps you!
>  
> •	Stream Deck + 800x480
> •	Keys are 120 x 120
>  
> Now the one thing is I don't believe that includes the touch strip as that would be it's own dimensions. They would be as followed:
>  
> •	Touch Strip is 800x100, but each of the four segments it's made up of is 200x100

Unfortunately, this didn't quite line up with the desired outcome. Using 800x480 provides what looks to be the correct outside dimensions, however each button appears to be > 120x120 – If viewed at any angle, you can see a section of pixels on all sides (not sure if the 120x120 is the ‘viewable area’, however the additional pixels are relevant to identifying the actual layout). I also believe that there is a buffer around the outside (i.e. the buttons aren’t as wide as the touch screen) which is also causing some issues. 

I'm hoping to get the exact required resolutions for this (possibly with 'bleed' areas to allow for indirect viewing), to allow me to create a seamless looking graphic. 

This is the graphic I created to test with: 
![image](https://github.com/elgatosf/elgato/assets/51277302/42a8c860-f701-442d-b6a1-3c29401690c0)

These are the outcome:
![Result-2](https://github.com/elgatosf/elgato/assets/51277302/5dd831fe-3f72-4dbf-ac61-aab334c90f0a)
![Result-3](https://github.com/elgatosf/elgato/assets/51277302/5f251188-ac18-47b9-b627-65036f90e219)
![Result-1](https://github.com/elgatosf/elgato/assets/51277302/cf61782b-0897-4943-9741-1005f3db6dcc)
![Result-4](https://github.com/elgatosf/elgato/assets/51277302/e87dfe93-dd42-4c8c-8648-1dd4965c13f2)

**Comment by GeekyEggo** [comment]
Pinging @tiptronic, he’s our resident images guru.

**Comment by tiptronic** [comment, marked as answer]
Hi @davewatson91 
your screenshots are already looking good. Please keep in mind the Stream Deck is made as a combination of LCD, housing, keys, etc... and therefore is NOT pixel-perfect.

Your dimensions also looking good (at least for Stream Deck plus). 
That being said: creating a pixel-perfect image which exactly fits the keys on every Stream Deck plus is likely not possible, because of the beforementioned manufacturing process.

120x120px is the safe-area of a key on the SD+ (internally Stream Deck uses 144x144px sizes, which work fine on all Stream Deck devices). If your target is just Stream Deck plus, you should add

- 4px of 'bleeding'  and 2px (mostly invisible) margin to each side...
I attached a sample image using this pattern... It's not perfect (especially if you depend on hard edges), but it should work on most Stream Deck plus devices.

![github_disc_85_p1](https://github.com/elgatosf/elgato/assets/80752/2ff5e69f-1a04-49f7-8adf-79cd7832c65d)

Another way (which will get better results) is, to cut out the desired portions from your source-image via code and use the SDK's `setImage` to place those bits onto a key. That way you can be sure there are no bleeding edges.

That's how we created the wallpapers on our old [KeyCreator](https://www.elgato.com/us/en/s/keycreator) (which doesn't support Stream Deck plus, however).

Here's a sample how that looks:
The portions taken had a width and height of `144px` and the spacing between the tiles is `44px` (The starting point in this example was/is the top-left edge of the first key (0,0).). That's what we found gives the best results on most Stream Deck devices (which share regular key-spacing, e.g. XL, mini, regular 15 key).

<img width="1137" alt="CleanShot_2023-12-18_21 02 59" src="https://github.com/elgatosf/elgato/assets/80752/909bef2e-9ff9-453c-a742-86c2fed20e17">

Hope that helps

**Reply by davewatson91** [reply]
Thanks @tiptronic - I've had a bit more of a play, and managed to get it looking pretty OK now - I appreciate the guidance! Enjoy your Christmas break 😄
